core/src/main/java/io/parsingdata/metal/Util.java

```diff
@@ -33,13 +33,13 @@
 
 public final class Util {
 
-    final private static char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray(); // Private because array content is mutable.
+    private static final char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray(); // Private because array content is mutable.
 
     private Util() {}
 
     public static <T>T checkNotNull(final T argument, final String name) {
         if (argument == null) {
-            throw new IllegalArgumentException("Argument " + name + " may not be null.");
+            throw new IllegalArgumentException(String.format("Argument %s may not be null.", name));
```

Comment:
Excellent point! I was considering replacing all string concatenation with this, but in that case maybe we need a small wrapper.

Reply:
I'm actually kind of torn on this `Locale.ROOT`-thing. I mean, the locale exists for a reason: to show things in people's correct local form.

So why would we want to force the localization to something? It can't be to make sure our unit-tests always run, because then we're basically telling users: you have to put up with badly formatted strings because our test strategy sucks...

Reply:
I think for the same reason that we _chose_ a language for our code, javadoc and readme. We are not going to include a translation for that, so why would we keep formatting of strings dynamic? That could also just confuse people. The language is English, but the formatting is Greek? It makes more sense to me to keep that in sync at all times, since we do not support multiple languages.

Reply:
That's a fair point! But then I still don't understand why we'd want to use `Locale.ROOT`, which is the empty locale? I mean, if we choose English for all our error messages and displaying everything, shouldn't we use `Locale.ENGLISH`?

Reply:
Yes! :D I never understood the use of ROOT anyway... ;)

Reply:
I added the helper method as 'Util.format()' that uses `Locale.ENGLISH`. I suppose the `ROOT` makes sense in a forensic context where you don't want localization to mangle the actual recovered value. In our situation now we only use this method for error messages, in which case we explicitly choose to do this in English.